src/pspm_transfer_function.m

```diff
@@ -39,9 +39,15 @@
 
 % check input arguments
 if nargin<1
-    errmsg='No data given.'; warning(errmsg);
+    warning('ID:invalid_input','No data given.'); return;
 elseif nargin<2
-    errmsg='No transfer constant given'; warning(errmsg);
+    warning('ID:invalid_input','No transfer constant given.'); return;
+elseif ~isnumeric(c)
+    warning('ID:invalid_input','The parameter ''c'' has to be numeric.'); return;
+elseif exist('Rs','var') && ~isnumeric(Rs)
+    warning('ID:invalid_input','The parameter ''Rs'' has to be numeric.'); return;
+elseif exist('offset','var') && ~isnumeric(offset)
+    warning('ID:invalid_input','The parameter ''offset'' has to be numeric.'); return;

```

Comment:
This works; but it makes the logic a bit hard to follow. I think it would be easier to maintain if we perform the argument validation after the `nargin` checks. This has two advantages:
1. We don't have to check for existence, because any variable that was not passed is already initialized with some default value.
2. `nargin` checks already do "number of parameters checks". Adding extra `exists` logic inside this if-elseif block makes the logic hard to follow. Avoiding this is beneficial for later debugging.

Reply:
Fair enough, it makes more sense !